samples/manage/azure-automation-automated-export/AutoExport.ps1

```diff
@@ -0,0 +1,280 @@
+		ToDrop,
+		Finished
+	}
+"@
+
+# The database and server pairs that will be exported.
+$databaseServerPairs =
+	@([pscustomobject]@{serverName="SAMPLESERVER1";databaseName="SAMPLEDATABASE1"},
+    [pscustomobject]@{serverName="SAMPLESERVER1";databaseName="SAMPLEDATABASE2"},
+    [pscustomobject]@{serverName="SAMPLESERVER2";databaseName="SAMPLEDATABASE3"});
+
+$serverCred = Get-AutomationPSCredential -Name 'NAMEOFSERVERCREDENTIAL1';
+$serverCred2 = Get-AutomationPSCredential -Name 'NAMEOFSERVERCREDENTIAL2';
+$serverCredentialsDictionary = @{'SAMPLESERVER1'=$serverCred;'SAMPLESERVER2'=$serverCred2}
+
+# The number of databases you want to have running at the same time.
+$batchingLimit = 10;
+# The number of times you want to retry if there is a failure.
+$retryLimit = 5;
+# The number of minutes you want to wait for an operation to finish before you fail.
+$waitInMinutes = 30;
+
+# This function takes the database and server names and creates a database object to use for the export.
+function CreateDatabaseObject($databaseName, $serverName)
+{
+    # Create the new object.
+	$dbObj = New-Object System.Object;
+    # Add the DatabaseName property and set it.
+    $dbObj | Add-Member -type NoteProperty -name DatabaseName -Value $databaseName;
+    # Add a unique time at the end of DatabaseCopyName so that we have a unique database name every time. 
+	$currentTime = Get-Date -format "_yyyy-MM-dd_HH:mm.ss";
+	$dbCopyName = $databaseName + $currentTime;
+    # Add the DatabaseCopyName property and set it.
+	$dbObj | Add-Member -type NoteProperty -name DatabaseCopyName -Value $dbCopyName;
+    # Add the ServerName property and set it.
+    $dbObj | Add-Member -type NoteProperty -name ServerName -Value $serverName;
+    # Add the Export property and set it to $null for now. This will be used to look up the export after it has been started.
+    $dbObj | Add-Member -type NoteProperty -name Export -Value $null;
+    # Add the DatabaseState property and set it to ToCopy so that the "state machine" knows to start the copy of the database.
+	$dbObj | Add-Member -type NoteProperty -name DatabaseState -Value ([DatabaseState]::ToCopy);
+    # Add the RetryCount property and set it to 0. This will be used to count the number of time we retry each failable operation.
+    $dbObj | Add-Member -type NoteProperty -name RetryCount -Value 0;
+    # Add the OperationStartTime property and set it to $null for now. This will be used when an operation starts to correcly do timeouts.
+    $dbObj | Add-Member -type NoteProperty -name OperationStartTime -Value $null;
+
+    # Return the newly created object.
+    return $dbObj;
+}
+
+# This function starts the copy of the database. If there is an error, we set the state to ToDrop. Otherwise, we set the state to Copying.
+function StartCopy($dbObj)
+{
+    # Start the copy of the database.
+	Start-AzureSqlDatabaseCopy -ServerName $dbObj.ServerName -DatabaseName $dbObj.DatabaseName -PartnerDatabase $dbObj.DatabaseCopyName;
```

Comment:
Ideally, we should use the ARM based powershell API, instead of the service management based API. This will most likely require to switch the authentication method as well.